pkg/cluster/cluster.go

```diff
@@ -234,6 +234,15 @@ func (c *Clusterer) Start(ctx context.Context) error {
 func (c *Clusterer) Stop() error {
 	switch node := c.Node.(type) {
 	case *GossipNode:
+		// The node is going away. We move to the Terminating state to signal
+		// that we should not be owners for write hashing operations anymore.
+		ctx, ccl := context.WithTimeout(context.Background(), 5*time.Second)
+		defer ccl()
+
+		err := node.ChangeState(ctx, peer.StateTerminating)
+		if err != nil {
+			level.Error(node.log).Log("msg", "failed to change state to Terminating before shutting down", "err", err)
+		}
```

Comment:
Should this happen earlier? i.e., should this happen before stopping the Flow controller?

Reply:
Can you add a TODO for us to investigate this?